### PR TITLE
Programs endpoint fixes

### PR DESCRIPTION
**Pull request by clintonb**
- Fixed tests
- Restored usage of `ProgramSerializer` for all calls to `/api/v1/programs/`

### CODE REVIEW

**Comment by clintonb** [comment]
@edx/ecommerce please review

File: course_discovery/apps/api/tests/test_serializers.py [hunk 1]
```diff
@@ -94,56 +94,95 @@ def test_invalid_data_user_create(self):
...
 
 
-class CourseSerializerTests(TestCase):
-    def test_data(self):
-        course = CourseFactory()
-        video = course.video
-
-        request = make_request()
+class MinimalCourseSerializerTests(TestCase):
+    maxDiff = None
```
Comment (MatthewPiatetsky) [inline course_discovery/apps/api/tests/test_serializers.py]: are all the maxDiff's necessary?

Comment (clintonb) [reply course_discovery/apps/api/tests/test_serializers.py]: Nope. Removing.

**Comment by MatthewPiatetsky** [comment]
Could you describe what fixes this PR is making?

**Comment by rlucioni** [comment]
@MatthewPiatetsky this PR uses full serializers when serializing programs, restoring all fields removed yesterday. It also fixes the tests which broke yesterday (the tests were using the wrong serializers).

👍

**Comment by MatthewPiatetsky** [comment]
@rlucioni what is it changing in addition to reverting yesterday's changes? if there is nothing else why not revert commit

**Comment by clintonb** [comment]
@MatthewPiatetsky the commit has value in the sense that, ultimately, we will be using the minimal serializers.

**Pull request merged**